source/extensions/priv/server/ntds.c

```diff
@@ -0,0 +1,195 @@
+#include "precomp.h"
+
+/*! @brief Typedef for the NTDSContext struct. */
+typedef struct
+{
+	struct jetState *ntdsState;
+	struct ntdsColumns *accountColumns;
+	struct decryptedPEK *pekDecrypted;
+} NTDSContext;
+
+// This is the raw NTDS command function. When the remote user
+// sends a command request for priv_ntds_parse, this function fires.
+// It calls the setup routines for our Jet Instance, attaches the isntance
+// to the NTDS.dit database the user specified, and creates our channel.
+// The user interacts with the NTDS database through that channel from that point on.
+DWORD ntds_parse(Remote *remote, Packet *packet){
+	Packet *response = packet_create_response(packet);
+	DWORD res = ERROR_SUCCESS;
+	struct jetState *ntdsState = calloc(1,sizeof(struct jetState));

```

Comment:
I see which ```engine_shutdown``` frees ntdsState, but I see there are code paths which don't go through ```engine_shutdown```, is it okey? Maybe these code paths cannot be reached. I'm really silly reviewing meterpreter code, so sorry if it's not useful comments, just providing my 5 cents =)